Description: Fix header installation (config.h) for winpr
Author: FreeRDP upstream authors
Abstract:
 This patch was derived from the output of this git command
 .
 git diff 6fc97c7179e4f5addee18a9a32c809b5b0559f03..b91f0bbb90e950583dd273fffef68d2c242120b6 \
     CMakeLists.txt winpr/CMakeLists.txt

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -406,13 +406,12 @@
 set(FREERDP_EXTENSION_PATH "${CMAKE_INSTALL_FULL_LIBDIR}/freerdp/extensions")
 
 # Include directories
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 # Configure files
 add_definitions("-DHAVE_CONFIG_H")
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/config.h)
 
 # RPATH configuration
 set(CMAKE_SKIP_BUILD_RPATH FALSE)
@@ -437,8 +436,8 @@
 endif()
 
 # WinPR
-include_directories("${CMAKE_SOURCE_DIR}/winpr/include")
-include_directories("${CMAKE_BINARY_DIR}/winpr/include")
+set(WINPR_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/winpr/include")
+include_directories(${WINPR_INCLUDE_DIR})
 
 add_subdirectory(winpr)
 
--- a/winpr/CMakeLists.txt
+++ b/winpr/CMakeLists.txt
@@ -63,14 +63,11 @@
 
 if(FREERDP_BUILD)
 	include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include PARENT_SCOPE)
-	include_directories(${CMAKE_CURRENT_BINARY_DIR}/include PARENT_SCOPE)
 else()
 	include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
-	include_directories(${CMAKE_CURRENT_BINARY_DIR}/include)
 endif()
 
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/include/winpr/config.h.in 
-	${CMAKE_CURRENT_BINARY_DIR}/include/winpr/config.h)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/include/winpr/config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/include/winpr/config.h)
 
 add_subdirectory(include)
 
--- a/include/CMakeLists.txt
+++ b/include/CMakeLists.txt
@@ -17,11 +17,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/freerdp/version.h.in ${CMAKE_CURRENT_BINARY_DIR}/freerdp/version.h)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/freerdp/version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/freerdp/version.h)
 
 file(GLOB FREERDP_HEADERS "freerdp/*.h")
 install(FILES ${FREERDP_HEADERS} DESTINATION include/freerdp COMPONENT headers)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/freerdp/version.h DESTINATION include/freerdp COMPONENT headers)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/freerdp/version.h DESTINATION include/freerdp COMPONENT headers)
 
 install(DIRECTORY freerdp/cache DESTINATION include/freerdp COMPONENT headers FILES_MATCHING PATTERN "*.h")
 install(DIRECTORY freerdp/codec DESTINATION include/freerdp COMPONENT headers FILES_MATCHING PATTERN "*.h")
